protocol/v2/ssv/validator/duty_executer.go

```diff
@@ -42,5 +42,9 @@ func (c *Committee) OnExecuteDuty(logger *zap.Logger, msg *types.EventMsg) error
 		return fmt.Errorf("could not start committee consume queue: %w", err)
 	}
 
+	if err := c.StopOldRunners(logger, executeDutyData.Duty.Slot); err != nil {
+		return fmt.Errorf("could not start committee duty: %w", err)
+	}
```

Comment:
Good point. Thanks
To prevent too frequent mtx lock I moved this to `Committee.StartDuty()` for cases when we create a new runner, a new goroutine is started to clean up the old runners.